commands/command_filter_process.go

```diff
@@ -40,25 +44,46 @@ func filterCommand(cmd *cobra.Command, args []string) {
 		ExitWithError(err)
 	}
 
-	_, err := s.NegotiateCapabilities()
+	caps, err := s.NegotiateCapabilities()
 	if err != nil {
 		ExitWithError(err)
 	}
 
+	var supportsDelay bool
+	for _, cap := range caps {
+		if cap == "capability=delay" {
+			supportsDelay = true
+			break
+		}
+	}
+
 	skip := filterSmudgeSkip || cfg.Os.Bool("GIT_LFS_SKIP_SMUDGE", false)
 	filter := filepathfilter.New(cfg.FetchIncludePaths(), cfg.FetchExcludePaths())
 
+	ptrs := make(map[string]*lfs.Pointer)
+
+	var q *tq.TransferQueue
+	closeOnce := new(sync.Once)
+	available := make(chan *tq.Transfer)
+
+	if supportsDelay {
+		q = tq.NewTransferQueue(tq.Download, getTransferManifest(), cfg.CurrentRemote)
+		go infiniteTransferBuffer(q, available)
+	}
+
 	var malformed []string
 	var malformedOnWindows []string
-
 	for s.Scan() {
 		var n int64
 		var err error
+		var delayed bool
 		var w *git.PktlineWriter
 
 		req := s.Request()
 
-		s.WriteStatus(statusFromErr(nil))
+		if !(req.Header["command"] == "smudge" && req.Header["can-delay"] == "1") && !(req.Header["command"] == "list_available_blobs") {
+			s.WriteStatus(statusFromErr(nil))
+		}
```

Comment:
Nit: I feel this makes the protocol a bit hard to read.
I would rather add `s.WriteStatus(statusFromErr(nil))` in the switch statement twice below.